tests/components/yandex_transport/test_yandex_transport_sensor.py

```diff
@@ -0,0 +1,117 @@
+"""Tests for the yandex transport platform."""
+
+import json
+import pytest
+
+import homeassistant.components.sensor as sensor
+import homeassistant.util.dt as dt_util
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+from tests.common import (
+    assert_setup_component,
+    async_setup_component,
+    MockDependency,
+    load_fixture,
+)
+
+REPLY = json.loads(load_fixture("yandex_transport_reply.json"))
+
+
+@pytest.fixture
+def mock_requester():
+    """Create a mock ya_ma module and YandexMapsRequester."""
+    with MockDependency("ya_ma") as ya_ma:
+        instance = ya_ma.YandexMapsRequester.return_value
+        instance.get_stop_info.return_value = REPLY
+        yield instance
+
+
+STOP_ID = 9639579
+ROUTES = ["194", "т36", "т47", "м10"]
+NAME = "test_name"
+TEST_CONFIG = {
+    "sensor": {
+        "platform": "yandex_transport",
+        "stop_id": 9639579,
+        "routes": ROUTES,
+        "name": NAME,
+    }
+}
+
+
+def true_filter(reply, filter_routes=None):
```

Comment:
Using this function, that is very similar to the code under test, to build the expected state that we use in the tests isn't optimal.

We're susceptible to mistakes in the logic driving the function, and a future change in the integration will need a change in this test function too.

We should try to be more declarative when describing the expected state. Can we just create a dict with the expected attributes which we save as a constant and another constant for the expected state value?

Reply:
sure